src/plot.cpp

```diff
@@ -980,16 +1019,552 @@ RGBColor random_color(void)
+              seg_color.red, seg_color.green, seg_color.blue);
+            double mixing = std::exp(-xs_[colormap_idx] * segments[i].length);
+            current_color =
+              current_color * mixing + (1.0 - mixing) * seg_color_vec;
+            RGBColor result;
+            result.red = static_cast<uint8_t>(current_color.x);
+            result.green = static_cast<uint8_t>(current_color.y);
+            result.blue = static_cast<uint8_t>(current_color.z);
+            data(horiz, vert) = result;
+          }
+
+          // Check to draw wireframe in horizontal direction. No inter-thread
+          // comm.
+          if (horiz > 0) {
+            if (!trackstack_equivalent(this_line_segments[tid][horiz],
+                  this_line_segments[tid][horiz - 1])) {
+              wireframe_initial(horiz, vert) = 1;
+            }
+          }
+        }
+      } // end "if" vert in correct range
+
+      // We require a barrier before comparing vertical neighbors' intersection
+      // stacks. i.e. all threads must be done with their line.
+#pragma omp barrier
+
+      // Now that the horizontal line has finished rendering, we can fill in
+      // wireframe entries that require comparison among all the threads. Hence
+      // the omp barrier being used. It has to be OUTSIDE any if blocks!
+      if (vert < pixels_[1]) {
+        // Loop over horizontal pixels, checking intersection stack of upper
+        // neighbor
+
+        const std::vector<std::vector<TrackSegment>>* top_cmp = nullptr;
+        if (tid == 0)
+          top_cmp = &old_segments;
+        else
+          top_cmp = &this_line_segments[tid - 1];
+
+        for (int horiz = 0; horiz < pixels_[0]; ++horiz) {
+          if (!trackstack_equivalent(
+                this_line_segments[tid][horiz], (*top_cmp)[horiz])) {
+            wireframe_initial(horiz, vert) = 1;
+          }
+        }
+      }
+
+      // We need another barrier to ensure threads don't proceed to modify their
+      // intersection stacks on that horizontal line while others are
+      // potentially still working on the above.
+#pragma omp barrier
+      vert += n_threads;
+    }
+  } // end omp parallel
+
+  // Now thicken the wireframe lines and apply them to our image
+  for (int vert = 0; vert < pixels_[1]; ++vert) {
+    for (int horiz = 0; horiz < pixels_[0]; ++horiz) {
+      if (wireframe_initial(horiz, vert)) {
+        if (wireframe_thickness_ == 1)
+          data(horiz, vert) = wireframe_color_;
+        for (int i = -wireframe_thickness_ / 2; i < wireframe_thickness_ / 2;
+             ++i)
+          for (int j = -wireframe_thickness_ / 2; j < wireframe_thickness_ / 2;
+               ++j)
+            if (i * i + j * j < wireframe_thickness_ * wireframe_thickness_) {
+              if (horiz + i >= 0 && horiz + i < pixels_[0] && vert + j >= 0 &&
+                  vert + j < pixels_[1])
+                data(horiz + i, vert + j) = wireframe_color_;
```

Comment:
There might be some redundant setting of values entries for wireframes on the boundary, but this is a cleaner to me. Feel free to take or leave it.

```suggestion
              int w_i = std::max(std::min(horiz + i, pixels_[0]), 0);
              int w_j = std::max(std::min(vert + j, pixels_[1]), 0);
              data(i, j) = wireframe_color_;
```

Reply:
Are we leaving this then, @gridley?

Reply:
eh, sure, I'll commit it! Your code is probably more idiomatic.